File: guides/micronaut-data-mongodb-synchronous/java/src/main/java/example/micronaut/FruitService.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;

import java.util.List;
import java.util.Optional;

interface FruitService {

    Iterable<Fruit> list();

    Fruit save(Fruit fruit);

    Optional<Fruit> find(@NonNull String id);

    Iterable<Fruit> findByNameInList(List<String> name);
}

File: guides/micronaut-data-mongodb-synchronous/python/src/example/micronaut/fruit_service.py
from abc import ABC, abstractmethod
from typing import List

from .fruit import Fruit


class FruitService(ABC):
    @abstractmethod
    def list(self) -> List[Fruit]:
        pass

    @abstractmethod
    def save(self, fruit: Fruit) -> Fruit:
        pass

    @abstractmethod
    def find(self, id: str) -> Fruit | None:
        pass

    @abstractmethod
    def find_by_name_in_list(self, names: List[str]) -> List[Fruit]:
        pass
